src/plone/restapi/services/relations/get.py

```diff
@@ -0,0 +1,326 @@
+        if relation.isBroken():
+            if not onlyBroken:
+                continue
+        else:
+            if onlyBroken:
+                continue
+        count += 1
+        if max and count > max:
+            break
+
+        source_obj = relation.from_object
+        target_obj = relation.to_object
+
+        if not unrestricted:
+            can_view = (not source_obj or checkPermission("View", source_obj)) and (
+                not target_obj or checkPermission("View", target_obj)
+            )
+            if not can_view:
+                continue
+
+        if onlyBroken:
+            results[relation.from_attribute].append(
+                [
+                    source_obj and source_obj.absolute_url() or "",
+                    target_obj and target_obj.absolute_url() or "",
+                ]
+            )
+        else:
+            results[relation.from_attribute].append(
+                {
+                    "source": make_summary(source_obj, request),
+                    "target": make_summary(target_obj, request),
+                }
+            )
+    return results
+
+
+def relation_stats():
+    if get_relations_stats is not None:
+        rels, broken = get_relations_stats()
+        results = {"relations": rels, "broken": broken}
+        return json_compatible(results)
+    else:
+        raise ImportError
+
+
+def getBrokenRelationNames():
+    relation_catalog = queryUtility(ICatalog)
+    if relation_catalog is None:
+        return []
+
+    relations = relation_stats()
+    return relations["broken"] and relations["broken"].keys() or []
+
+
+def getStaticCatalogVocabularyQuery(vocabularyFactoryName):
+    factory = queryUtility(IVocabularyFactory, vocabularyFactoryName)
+    if factory:
+        return factory().query
+    return
+
+
+class GetRelations(Service):
+    """Get relations or stats
+
+        source: UID of content item
+        target: UID of content item
+        relation: name of a relation
+        onlyBroken: boolean: dictionary with broken relations per relation type
+        max: integer: maximum of results
+
+    Returns:
+        stats if no parameter, else relations
+    """
+
+    def __init__(self, context, request):
+        super().__init__(context, request)
+        self.sm = getSecurityManager()
+
+    def reply(self):
+        # Disable CSRF protection
+        if "IDisableCSRFProtection" in dir(plone.protect.interfaces):
+            alsoProvides(self.request, plone.protect.interfaces.IDisableCSRFProtection)
+
+        source = self.request.get("source", None)
+        target = self.request.get("target", None)
+        relationship = self.request.get("relation", None)
+        rebuild = self.request.get("rebuild", False)
+        max = self.request.get("max", False)
+        onlyBroken = self.request.get("onlyBroken", False)
+        query_source = self.request.get("query_source", None)
+        query_target = self.request.get("query_target", None)
```

Comment:
Some of these parameters are not mentioned in the docs.

Reply:
Documentation is updated with these options. I kept the description short, cause these options are for specific cases and I prefer to keep the focus on the main options.